### PR TITLE
Add a user document for the scheduling framework

### PR DESCRIPTION
**Pull request by bsalamat**
Add documentation on the scheduling framework and how to use it.

ref kubernetes/enhancements#624

cc/ @ahg-g
/sig scheduling

### CODE REVIEW

File: content/en/docs/concepts/configuration/scheduling-framework.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    If any permit plugin denies a pod, it is returned to the scheduling queue.
+    This will trigger [Un-reserve](#un-reserve) plugins.
+
+1.  **wait** (with a timeout) \
+    If a permit plugin returns "wait", then the pod is kept in the permit phase
+    until a [plugin approves it](#frameworkhandle). If a timeout occurs, **wait**
+    becomes **deny** and the pod is returned to the scheduling queue, triggering
+    [un-reserve](#un-reserve) plugins.
+
+**Approving a pod binding**
```
Comment (ahg-g) [inline content/en/docs/concepts/configuration/scheduling-framework.md]: Approve a waiting pod?

File: content/en/docs/concepts/configuration/scheduling-framework.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+While any plugin can receive the list of reserved pods from the cache and
```
Comment (ahg-g) [inline content/en/docs/concepts/configuration/scheduling-framework.md]: s/receive/access

s/reserved/"waiting"

File: content/en/docs/concepts/configuration/scheduling-framework.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+approve them (see [`FrameworkHandle`](#frameworkhandle)) we expect only the permit
+plugins to approve binding of reserved Pods that are in "waiting" state. Once a
+pod is approved, it is sent to the pre-bind phase.
```
Comment (ahg-g) [inline content/en/docs/concepts/configuration/scheduling-framework.md]: nit: sometimes the "P" in pod is capitalized, sometimes it is not, perhaps be consistent with one or the other.

**Comment by makoscafee** [comment]
Hi, @bsalamat. Ping, you need help for anything to help speed up this work? We are looking to have all PR for release 1.15 merged by June 10th. Let me know if there is anything docs team can assist on this.

File: content/en/docs/concepts/configuration/scheduling-framework.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+## Scheduling Cycle & Binding Cycle
+
+The scheduling cycle selects a node for the pod, and the binding cycle applies
+that decision to the cluster. Together, a scheduling cycle and binding cycle are
+referred to as a "scheduling context".
+
+Scheduling cycles are run serially, while binding cycles may run concurrently.
+
+A scheduling cycle or binding cycle can be aborted if the pod is determined to
```
[the remark this answers is not in the document]
Comment (bsalamat) [reply content/en/docs/concepts/configuration/scheduling-framework.md]: done

File: content/en/docs/concepts/configuration/scheduling-framework.md (hunk 1 above)
Comment (bsalamat) [reply content/en/docs/concepts/configuration/scheduling-framework.md]: It is actually approving a waiting pod for binding. I guess approving a pod binding is more accurate.

File: content/en/docs/concepts/configuration/scheduling-framework.md (hunk 2 above)
Comment (bsalamat) [reply content/en/docs/concepts/configuration/scheduling-framework.md]: done

File: content/en/docs/concepts/configuration/scheduling-framework.md (hunk 3 above)
Comment (bsalamat) [reply content/en/docs/concepts/configuration/scheduling-framework.md]: Good point. In user docs we usually capitalize them. I updated the doc.

**Comment by bsalamat** [comment]
@makoscafee This is ready to be merged if we can get it approved.

**Comment by WanLinghao** [comment]
@bsalamat  hi, I am confused about what is the "default plugin" when I read the content about disable/enable plugin. Could you please improve this page against this. Thank you

**Pull request merged**